packages/nouns-contracts/contracts/governance/NounsDAOLogicV2.sol

```diff
@@ -0,0 +1,895 @@
+        );
+
+        uint16 oldMaxQuorumVotesBPS = params.maxQuorumVotesBPS;
+        params.maxQuorumVotesBPS = newMaxQuorumVotesBPS;
+
+        _writeQuorumParamsCheckpoint(params);
+
+        emit MaxQuorumVotesBPSSet(oldMaxQuorumVotesBPS, newMaxQuorumVotesBPS);
+    }
+
+    function _setQuorumVotesBPSOffset(uint16 newQuorumVotesBPSOffset) external {
+        require(msg.sender == admin, 'NounsDAO::_setQuorumVotesBPSOffset: admin only');
+        DynamicQuorumParams memory params = getDynamicQuorumParamsAt(block.number);
+
+        uint16 oldQuorumVotesBPSOffset = params.quorumVotesBPSOffset;
+        params.quorumVotesBPSOffset = newQuorumVotesBPSOffset;
+
+        _writeQuorumParamsCheckpoint(params);
+
+        emit QuorumVotesBPSOffsetSet(oldQuorumVotesBPSOffset, newQuorumVotesBPSOffset);
+    }
+
+    function _setQuorumLinearCoef(uint32 newQuorumLinearCoef) external {
+        require(msg.sender == admin, 'NounsDAO::_setQuorumLinearCoef: admin only');
+        DynamicQuorumParams memory params = getDynamicQuorumParamsAt(block.number);
+
+        uint32 oldQuorumLinearCoef = params.quorumLinearCoef;
+        params.quorumLinearCoef = newQuorumLinearCoef;
+
+        _writeQuorumParamsCheckpoint(params);
+
+        emit QuorumLinearCoefSet(oldQuorumLinearCoef, newQuorumLinearCoef);
+    }
+
+    function _setQuorumQuadraticCoef(uint32 newQuorumQuadraticCoef) external {
+        require(msg.sender == admin, 'NounsDAO::_setQuorumQuadraticCoef: admin only');
+        DynamicQuorumParams memory params = getDynamicQuorumParamsAt(block.number);
+
+        uint32 oldQuorumQuadraticCoef = params.quorumQuadraticCoef;
+        params.quorumQuadraticCoef = newQuorumQuadraticCoef;
+
+        _writeQuorumParamsCheckpoint(params);
+
+        emit QuorumQuadraticCoefSet(oldQuorumQuadraticCoef, newQuorumQuadraticCoef);
+    }
+
+    function _setDynamicQuorumParams(DynamicQuorumParams memory params) public {
+        require(msg.sender == admin, 'NounsDAO::_setDynamicQuorumParams: admin only');
+        require(
+            params.minQuorumVotesBPS >= MIN_QUORUM_VOTES_BPS_LOWER_BOUND &&
+                params.minQuorumVotesBPS <= MIN_QUORUM_VOTES_BPS_UPPER_BOUND,
+            'NounsDAO::_setDynamicQuorumParams: invalid min quorum votes bps'
+        );
+        require(
+            params.maxQuorumVotesBPS <= MAX_QUORUM_VOTES_BPS_UPPER_BOUND,
+            'NounsDAO::_setDynamicQuorumParams: invalid max quorum votes bps'
+        );
+        require(
+            params.minQuorumVotesBPS <= params.maxQuorumVotesBPS,
+            'NounsDAO::_setDynamicQuorumParams: min quorum votes bps greater than max'
+        );
+
+        _writeQuorumParamsCheckpoint(params);
+
+        emit DynamicQuorumParamsSet(
+            params.minQuorumVotesBPS,
+            params.maxQuorumVotesBPS,
+            params.quorumVotesBPSOffset,
+            params.quorumLinearCoef,
+            params.quorumQuadraticCoef
+        );
```

Comment:
Thinking we could just reuse the above events for the aggregate function. Nice to only have a single event for the update of each dynamic quorum parameter.